Update zboss libraries to v3.5.2.0 (#457)
nrfconnect/sdk-nrfxlib: zboss/doc/zboss_configuration.rst

   :depth: 2

The ZBOSS stack is distributed as a set of precompiled linkable libraries.

> The library support different Zigbee device roles, each variant having its own configuration.

Reviewer: ```suggestion
The libraries support different Zigbee device roles, with each variant having its own configuration.
```

Author: Proposed change applied.

This allows you to scale the application and select the most suitable set of features.

> ZBOSS libraries

Reviewer: ZBOSS device role libraries? Or maybe we don't need this section header at all?

Author: removed

***************************

|list_intro|

* :file:`zboss/include/osif/libzboss_config.h` - library for Coordinators and Routers
* :file:`zboss/include/osif/libzboss_config.ed.h` - library for End Devices

|list_outro|

You can enable additional features in the ZBOSS libraries using the following Kconfig options:

> * :option:`CONFIG_ZIGBEE_LIBRARY_NCP_DEV` - With this option enabled, the application may link NCP commands implementation.

Reviewer: ```suggestion
* :option:`CONFIG_ZIGBEE_LIBRARY_NCP_DEV` - With this option enabled, the application includes the implementation of NCP commands.
```

Author: Proposed change applied.

ZBOSS library certification
***************************

> Zigbee ZBOSS libraries in |NCS| are always tested in-house by Nordic Semiconductor to conform to Zigbee Pro R22 test specification.

Reviewer: ```suggestion
Zigbee ZBOSS libraries in the |NCS| are always tested in-house by Nordic Semiconductor to conform to Zigbee Pro R22 test specification.
```

However, these libraries are only periodically verified by external test service provider and granted Zigbee Compliant Platform certification by Zigbee Alliance.

> .. |list_intro| replace:: For a complete list of the ZBOSS configuration options, see the following files:

Reviewer: Please change |list_outro| to the following:
``These libraries are used in the Zigbee protocol configuration in the |NCS| when defining the Zigbee device role, as described in :ref:`nrf:zigbee_ug_configuration` in the |NCS| documentation.``

Author: Proposed change applied.
